### ISSUE TITLE
Output only most specific term IBA in GAF generation

### ISSUE THREAD

**Issue by dustine32** [pantherdb/fullgo_paint_update#56]
Here's an example of two IBDs that can produce redundant IBAs.

These IBDs in PTHR48013 produce IBAs for both terms to leaf `HUMAN|HGNC=6847|UniProtKB=O14733`:
`PTN000684494 - GO:0000165`
`PTN001220492 - GO:0051403`

Term GO:0051403 is a child of GO:0000165:
![image](https://user-images.githubusercontent.com/2678599/189182063-21b8c07d-7973-4cf5-81fa-4723c8551786.png)

Node `PTN000684494` is an ancestor of `PTN001220492` and has the more generic term, so this arrangement makes sense and these IBDs are not really redundant. But there is still a bug in the IBA perl script [createGAF.pl](https://github.com/pantherdb/fullgo_paint_update/blob/master/scripts/createGAF.pl) and the fix is to simply keep only the most specific term IBA for a node (e.g., human `UniProtKB:O14733`) if the multiple IBA terms are related.

**Comment by mugitty** [comment]
The tree propagation logic also needs to be updated to stop propagating the less specific term.

**Comment by dustine32** [comment]
Need to look at NOT IBA examples to ensure they are being handled correctly. If there are two IBAs to the same gene where the two terms are related, the more generic term should be retained/written out. Example case:
```
MGI:MGI:1921818 NOT|enables GO:0004674 IBA
MGI:MGI:1921818 NOT|enables GO:0004706 IBA
```
Term GO:0004674 is more generic so the GO:0004706 NOT IBA should be removed:
![image](https://user-images.githubusercontent.com/2678599/194119384-2c2691e2-3250-49e1-a7d6-1bd97f307115.png)

**Comment by dustine32** [comment]
This change has been implemented and used for the 17.0/2022-11-04 IBA GAF release.
